### DISCUSSION TITLE
Profiling, where do I begin

### DISCUSSION THREAD

**Discussion by MikePresman** [typescript-language-server/typescript-language-server#907] [General]
Hi everyone, I'm curious where would one begin in profiling a slow LSP in my codebase?
What tools, techniques, etc have been helpful for any of you?
Thanks !

**Comment by MikePresman** [comment]
found this so far: https://github.com/microsoft/TypeScript-wiki/blob/main/Debugging-Language-Service-in-VS-Code.md

**Reply by rchl** [reply]
The approach with using `TSS_DEBUG` or `TSS_DEBUG_BRK` should work with this code base too but maybe it's enough to just enable tsserver logging (`initializationOptions.tsserver.logVerbosity`) and inspect the log manually? It should allow you to see which requests are slow. But typically slowness is due to large number of files exposed to tsserver or very complex types. That would likely also show up as slow compilation using `tsc` on the command line.
